src/parallel/job_token.rs

```diff
@@ -163,7 +163,7 @@ mod inherited_jobserver {
 
     pub(crate) struct ActiveJobServer<'a> {
         jobserver: &'a JobServer,
-        helper_thread: OnceCell<HelperThread>,
+        helper_thread: OnceCell<Option<HelperThread>>,
```

Comment:
I think we're losing the plot here. `OnceLock` is already spiritually an `Option`, where the "Some or None?" discriminant is encoded using the `Once`'s internal "completed or not?" state. We shouldn't need all this.

Reply:
Same as https://github.com/rust-lang/cc-rs/pull/1154#discussion_r1676984166

> Because the helper thread creation might fail, and since I can't implement get_or_try_init in a few lines, I put an Option there instead for the error.

Reply:
Then why not just put everything inline?

Reply:
> Then why not just put everything inline?

I'd like to make the `OnceLock` implementation as simple as possible (to make it easier to audit).

If I was to move this to `get_or_try_init`, then I'm not sure how to handle this case:
 - `get_or_try_init` error
 - `get_or_try_init` gets called again, but `Once` is already completed

I could put that `Option` inline, then I would have to return `Result<Option<T>, E>`, sounds a bit more complex than just `&T` in `get_or_init`

Reply:
My view is simplicity is achieved by avoiding genericity over cases that we don't actually want to reason about or handle properly. We don't necessarily need OnceLock<HelperThread>, we need OnceHelperThread.

Reply:
Hmmm that sounds reasonable, I could add a `OnceHelperThread`

Reply:
Took another look, perhaps we could use an `Option` for helper thread, since it is created for each call to `Build::compile_objects`, so it can have exclusive access to the struct.

Reply:
Certainly!

Reply:
Updated, now it uses an `Option` instead of `OnceLock`, it's much simpler and has less overhead!